### PR TITLE
Split up pico_ice_sdk libraries

### PR DESCRIPTION
**Pull request by alastairpatrick**

### CODE REVIEW

**Review by josuah — COMMENTED** [review]
Having a directory named `include` feels like "this directory will be passed as an `-I` flag.

So having a `-I./include/ice_usb` could be misleading. What about and `include_usb` then?

This might look a bit less aesthetic than just `include`, probably why pico-sdk went with `$module/include/$module.h` (with even an extra level on top of $module though).

File: example/pico-ssram/firmware.c [hunk 1]
```diff
@@ -23,20 +20,14 @@ void write_callback(void* context) {
...
 
-    // Dont let the FPGA on the bus so we get exclusive access
-    ice_fpga_halt();
-
     for (uint16_t i = 0; i < DATA_LEN; i++) {
         write_data[i] = i;
     }
 
     for (;;) {
-        tud_task();
```
Comment (josuah) [inline example/pico-ssram/firmware.c:38]: not sure I follow: is this related/necessary or a bonus?

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -1,32 +1,66 @@
...
     )
+    target_include_directories(pico_ice_usb INTERFACE ../include)
+    target_link_libraries(pico_ice_usb INTERFACE
+        hardware_uart
+        tinyusb_device
+        tinyusb_board
+    )
 
-    target_include_directories(pico_ice_sdk INTERFACE ../include)
+    #### pico_ice_sdk ####
```
Comment (josuah) [inline src/CMakeLists.txt:53]: nice way to do it: pico_ice_sdk as a "do-it-all" with defaults (and then we can be more aggressive here), with pico_ice_[...] that it contains as the sub-part of it.

**Comment by alastairpatrick** [comment]
Alternative patch landed: https://github.com/tinyvision-ai-inc/pico-ice-sdk/commit/4aea79e0e136b7fa08eb1c277a66755161207053
